src/gravel/controllers/orch/nfs.py

```diff
@@ -0,0 +1,100 @@
+# project aquarium's backend
+# Copyright (C) 2021 SUSE, LLC.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+from typing import Any, Dict, List, Optional
+
+from pydantic.tools import parse_obj_as
+from pydantic import BaseModel
+
+from gravel.controllers.orch.ceph import CephCommandError, Mgr
+
+
+class NFSDaemonModel(BaseModel):
+    hostname: str
+    ip: List[str]
+    port: int
+
+
+class NFSServiceModel(BaseModel):
+    name: str
+    daemons: List[NFSDaemonModel]
+
+
+class NFSError(Exception):
+    pass
+
+
+class NFSService:
+    mgr: Mgr
+
+    def __init__(self):
+        self.mgr = Mgr()
+
+    def _call(self, cmd: Dict[str, Any]) -> Any:
+        try:
+            return self.mgr.call(cmd)
+        except CephCommandError as e:
+            raise NFSError(e) from e
+
+    def create(self, name: str, placement: Optional[str]) -> str:
+        cmd = {
+            'prefix': 'nfs cluster create',
+            'type': 'cephfs',  # TODO: fix upstream
+            'clusterid': name,
+        }
+        if placement:
+            cmd['placement'] = placement
+        return self._call(cmd)['result']
+
+    def update(self, name: str, placement: str) -> str:
```

Comment:
Empty placement means let the ceph orchestrator decide.

For NFS this would be a single daemon for the entire cluster:
https://github.com/ceph/ceph/blob/c707d451fbb5528c2384c9fbc4356b7b112366e2/src/pybind/mgr/cephadm/module.py#L2128

For aquarium I think we should default to a daemon per node: `placement: '*'`. See also the `ServiceRequest` model in this PR.

Reply:
Alright, just checking. Thanks. :)